workos/_base_client.py

```diff
@@ -13,37 +17,98 @@
 from workos.webhooks import WebhooksModule
 
 
-class BaseClient(Protocol):
+HTTPClientType = TypeVar("HTTPClientType", bound=HTTPClient)
+
+
+class BaseClient(Generic[HTTPClientType]):
     """Base client for accessing the WorkOS feature set."""
 
+    _api_key: str
+    _base_url: str
+    _client_id: str
+    _request_timeout: int
     _http_client: HTTPClient
 
+    def __init__(
+        self,
+        *,
+        api_key: Optional[str],
+        client_id: Optional[str],
+        base_url: Optional[str] = None,
+        request_timeout: Optional[int] = None,
+        http_client_cls: Type[HTTPClientType],
+    ) -> None:
+        api_key = api_key or os.getenv("WORKOS_API_KEY")
+        if api_key is None:
+            raise ValueError(
+                "WorkOS API key must be provided when instantiating the client or via the WORKOS_API_KEY environment variable."
+            )
+
+        self._api_key = api_key
+
+        client_id = client_id or os.getenv("WORKOS_CLIENT_ID")
+        if client_id is None:
```

Comment:
I don't think all of our APIs require a `client_id`, should we always require it? The Node SDK does not throw for a missing `client_id`